client/device/duoadminpanel/create_test.go

```diff
@@ -0,0 +1,136 @@
+package duoadminpanel_test
+
+import (
+	"context"
+	"github.com/CiscoDevnet/terraform-provider-cdo/go-client/device/duoadminpanel"
+	"github.com/CiscoDevnet/terraform-provider-cdo/go-client/internal/publicapi/transaction"
+	"github.com/CiscoDevnet/terraform-provider-cdo/go-client/internal/publicapi/transaction/transactionstatus"
+	"github.com/CiscoDevnet/terraform-provider-cdo/go-client/internal/publicapi/transaction/transactiontype"
+	"github.com/CiscoDevnet/terraform-provider-cdo/go-client/internal/url"
+	"github.com/CiscoDevnet/terraform-provider-cdo/go-client/model/device/tags"
+	"github.com/stretchr/testify/assert"
+	"net/http"
+	"testing"
+	"time"
+
+	internalHttp "github.com/CiscoDevnet/terraform-provider-cdo/go-client/internal/http"
+	"github.com/jarcoal/httpmock"
+)
+
+func TestDuoAdminPanelCreate(t *testing.T) {
+	httpmock.Activate()
+	defer httpmock.DeactivateAndReset()
+
+	createInput := duoadminpanel.CreateInput{
+		Name:           "test-name",
+		Host:           "test-host",
+		IntegrationKey: "test-int-key",
+		SecretKey:      "test-secret-key",
+		Labels:         []string{"lab1", "lab2", "lab3"},
+	}
+
+	doneTransaction := transaction.Type{
+		TransactionUid:        "test-TransactionUid",
+		TenantUid:             "test-TenantUid",
+		EntityUid:             "test-EntityUid",
+		EntityUrl:             "test-EntityUrl",
+		TransactionPollingUrl: "test-TransactionPollingUrl",
+		SubmissionTime:        "test-SubmissionTime",
+		LastUpdatedTime:       "test-LastUpdatedTime",
+		Type:                  transactiontype.ONBOARD_DUO_ADMIN_PANEL,
+		Status:                transactionstatus.DONE,
+	}
+
+	errorTransaction := transaction.Type{
+		TransactionUid:        "test-TransactionUid",
+		TenantUid:             "test-TenantUid",
+		EntityUid:             "test-EntityUid",
+		EntityUrl:             "test-EntityUrl",
+		TransactionPollingUrl: "test-TransactionPollingUrl",
+		SubmissionTime:        "test-SubmissionTime",
+		LastUpdatedTime:       "test-LastUpdatedTime",
+		Type:                  transactiontype.ONBOARD_DUO_ADMIN_PANEL,
+		Status:                transactionstatus.ERROR,
+		ErrorMessage:          "test-ErrorMessage",
+		ErrorDetails: map[string]string{
+			"test-key": "test-details",
+		},
+	}
+
+	createdDevice := duoadminpanel.ReadOutput{
+		Uid:  doneTransaction.EntityUid,
+		Name: createInput.Name,
+		Tags: tags.New(createInput.Labels...),
+	}
+
+	expectedCreateOutput := duoadminpanel.CreateOutput{
+		Uid:   createdDevice.Uid,
+		Name:  createdDevice.Name,
+		State: createdDevice.State,
+		Tags:  createdDevice.Tags,
+	}
+
+	baseUrl := "https://test.com"
```

Comment:
Lets use a cisco owned domain - just in case a call was ever to go out. E.g. just "https://test.cisco.com"

Reply:
yea make sense

Reply:
done 👍